csharp/sample/Microsoft.ML.OnnxTraining.MNISTSample/BinaryFile.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) 2018-2020 SignalPop LLC, and contributors. All rights reserved.
+// License: Apache 2.0
+// License: https://github.com/MyCaffe/MyCaffe/blob/master/LICENSE
+// Modified from Original Source: https://github.com/MyCaffe/MyCaffe/blob/master/MyCaffe.data/MnistDataLoader.cs
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+/// <summary>
+/// The MyCaffe.data namespace contains dataset creators used to create common testing datasets such as MNIST, CIFAR-10 and VOC0712.
+/// </summary>
+namespace MyCaffe.data
+{
+    /// <summary>
+    /// The BinaryFile class is used to manage binary files used by the MNIST dataset creator.
+    /// </summary>
+    class BinaryFile : IDisposable
+    {
+        FileStream m_file;
+        BinaryReader m_reader;
+
+        /// <summary>
+        /// The constructor.
+        /// </summary>
+        /// <param name="strFile">Specifies the filename.</param>
+        public BinaryFile(string strFile)
+        {
+            m_file = File.Open(strFile, FileMode.Open, FileAccess.Read, FileShare.Read);
+            m_reader = new BinaryReader(m_file);
+        }
+
+        /// <summary>
+        /// Release all resources.
+        /// </summary>
+        public void Dispose()
+        {
+            m_reader.Close();
+        }
```

Comment:
You need to follow MS standards for implementing IDisposable().

Reply:
Updated.

Reply:
You need to suppress the finalizer and also add the finalizer. I know everything about finalizers, but these are the rules.

---
In reply to: [544544162](https://github.com/microsoft/onnxruntime/pull/6141#discussion_r544544162) [](ancestors = 544544162)

Reply:
Updated.